modules/Library/library_lending_item_edit.php

```diff
@@ -20,15 +20,21 @@
 use Gibbon\Forms\Form;
 use Gibbon\Forms\DatabaseFormFactory;
 
+$gibbonLibraryItemEventID = $_GET['gibbonLibraryItemEventID'] ?? null;
```

Comment:
Sure.

I was translating the code to the way it actually works now. But if there is an empty string check down the road, it'd probably best to use empty string instead of `null`.

Reply:
It seems logically equivalent if I translate the `== ''` checks to `empty()` checks.

Reply:
It _is_ a refactoring effort, so it's often not just a case of changing the obvious `== ''` checks to `empty()` 😬 For the sake of these PRs though, it may help to keep the changes confined to one purpose.

In the case of $GET variables, they can also be empty in the url itself (`&foo` or `&foo=`), in which case PHP sets them to an empty string. I think the default variable value should match this behaviour, so that the type-whether set or empty-is still a string and not null.

Reply:
I understand that they are not always logically equivalent. But for the 4 cases that we're talking about in this module, they are all expecting non-empty string to these variables. So both of these cases:
* `$_GET['foo'] === ''`
* `!isset($_GET['foo'])`
are supposed to counted as error in the check.

That's why I said it's logically equivalent in our case to convert to a `empty()` check.

Reply:
Yep, I agree `empty()` is logically equivalent. This is the case of different programmers and different coding styles, and also the transitional nature of refactoring 😄 I do tend to change them in the codebase when I spot them.

> which is how Ross has coded them in the past, and I tend to use !empty() so the scripts are a bit transitional

However, for the variable themselves I believe the null-coalesce should still be a `?? ''` and not `?? null`, for the above reasons:

> In the case of $_GET variables, they can also be empty in the url itself (&foo or &foo=), in which case PHP sets them to an empty string. I think the default variable value should match this behaviour, so that the type-whether set or empty-is still a string and not null.

That was the type remains consistent, and there's two states: strong or empty string (without an additional null state).

Reply:
I tend to leave unset variables a `null` default to signify that they're not set. In case we want more granulate handling to the variable state (i.e. different error message for "not set" than "empty"), we'd have more information to do so.

Just a personal style. If you insist a `?? ''`, I would change it you way.

Reply:
Ah, I must be in a different camp 😉 (at lease for this case!)

https://www.lucidchart.com/techblog/2015/08/31/the-worst-mistake-of-computer-science/

Reply:
LOL.

I'll change it then.

Reply:
Done.